ERCS/erc-7730.md

```diff
@@ -0,0 +1,1033 @@
+
+### `Context` section
+
+The `context` section describes a set of *constraints* that must be verified by the structured data and container structure before formatting them using the ERC-7730 file. A wallet MUST verify that the structured data and container it is trying to sign matches the constraints of the `context` section.
+
+The current version of this specification only supports two types of binding context, EVM smart contracts and EIP-712 domains. 
+
+All context support an `$id` sub-key as an internal identifier (only relevant to provide a human readable name to the ERC-7730 file)
+
+#### EVM smart contract binding context (denoted 'calldata')
+
+The `contract` sub-key is used to introduce an EVM smart contract binding context, with the following constraints expressed as sub-keys of `contract`.
+  
+**`contract.abi`**
+
+Either an URL of the reference ABI (served in json representation), or the ABI itself as a json array.
+
+A wallet MUST verify that the ABI is applicable to the contract being called.
+
+All paths described in the ERC-7730 starting with the `#.` root node (typically used to describe a single parameter of the contract call) are using selectors names coming from the parameter names of the ABI referenced.
+
+The `contract.abi` key is mandatory for a smartcontract ERC-7730 file.
+
+**`contract.deployments`**
+
+An array of deployments options. Wallets MUST verify that the target chain and contract address of the containing transaction MUST match one of these deployment options.
+
+A deployment option is an object with:
+  * `chainId`: an [EIP-155 identifier](./eip-155.md) of the chain the descibed contract is deployed on.
+  * `address`: the address of the deployed contract on specified `chainId` chain.
+
+---
+*The following constraints for contracts are considered draft*
+
+**`contract.addressMatcher`**
+
+An URL that can be used to check whether the current transaction `(chainId, contract address)` can be formatted using the ERC-7730 file.
+
+A wallet MAY use this URL to check unknown addresses, default is to fail constraint on unknown addresses.
+
+**`contract.factory`**
+
+An object describing the factory used to deploy smartcontracts that can be clear signed using the ERC-7730 file.
+
+A factory is a json object with:
+* `deployEvent` key, specifying the solidity signature of the events emitted when deploying a clear-signable contract.
+* `deployments` key: an array of deployment options as in `contract.deployments`. These deployments represent the addresses at which the *factory* is deployed.  
+
+To verify a factory constraints a wallet MUST check that:
+* The current transaction destination address is included in an event of signature `factory.deployEvent`
+* The emitter of the event is a factory contract deployed at an address matching one of the deployment option in `factory.deployments`
+---
+
+*Examples*
+
+[TBD]
```

Comment:
If you use HTML-style comments, the linter will make sure you replace them before moving to Review:

```suggestion
<!-- TBD -->
```